### DISCUSSION TITLE
General question usage

### DISCUSSION THREAD

**Discussion by madelaney** [korandoru/hawkeye#135] [Q&A] [answered]
On Mac OS, using v5.3.1, I'm running into an issue where I always get the following error:
```console
Error: InvalidConfig { message: "no header source found in config", loc: Location { file: "/Users/mdelaney/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hawkeye-fmt-5.3.1/src/processor.rs", line: 74, column: 64 } }
Exit 1
```

I'm invoking using `hawkeye format --config licenserc.toml` and the following gist as the config - https://gist.github.com/madelaney/92a3d6c11ff7b699a8f32db7216f5ee0

Everything seems to be there but I'm not sure why it's failing.

**Comment by tisonkun** [comment, marked as answer]
This is a bug.

Sorry, I missed this path during the migration.

Here is a fix - https://github.com/korandoru/hawkeye/pull/136.

**Reply by tisonkun** [reply]
[v5.3.2](https://github.com/korandoru/hawkeye/releases/tag/v5.3.2) released with this supported.

@madelaney This seems a template of BSD. I wonder if a -

```
headerID = "BSD-3-Clause" # SPDX ID
```

a good experience for you.

**Reply by madelaney** [reply]
Awesome, I pulled the new version and it works as expected. Thank you for the quick turnaround. ☕

**Comment by madelaney** [comment]
> [v5.3.2](https://github.com/korandoru/hawkeye/releases/tag/v5.3.2) released with this supported.
> 
> @madelaney This seems a template of BSD. I wonder if a -
> 
> ```
> headerID = "BSD-3-Clause" # SPDX ID
> ```
> 
> a good experience for you.

Sorry, I don't understand the context. Could you explain more?

**Reply by tisonkun** [reply]
Currently, we have two methos to configure the content of header template: `inlineHeader` and `headerPath`.

We bundled Apache-2.0.txt and Apache-2.0-ASF.txt so that `headerPath = "Apache-2.0.txt"` can match even if users do not have such file in workdir.

Your situation seems suitable to add a new bundled "BSD-3-Clause.txt" file. And I wonder if we should introduce a new concept "headerId".

After some more consideration, I prefer to keep use the `headerPath` with bundled ones way. So if you'd like to make BSD-3-Clause a bundled template, you can add it to https://github.com/korandoru/hawkeye/tree/main/fmt/src/license and modify:

https://github.com/korandoru/hawkeye/blob/6a159b1e6a35f6bf81d04cb65531ba82cbc8e50a/fmt/src/license/mod.rs#L55-L65

Also I think the bundled ones should take lower priority to user configured ones.
